File: list/leetcode206.cpp
#include <iostream>
#include "list_node.h"

using namespace std;

ListNode* reverseList(ListNode* head){
    ListNode *prev = nullptr;
    ListNode *cur = head;
    while (cur != nullptr){
        ListNode *tmp = cur;
        cur = cur->next;
        tmp->next = prev;
        prev = tmp;
    }
    return prev;
}

int main(){
    vector<int> nums = {};
    ListNode *head = make_list(nums);
    ListNode *re = reverseList(head);
    system("pause");
    return 0;
}

File: list/leetcode206.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
"""
    @File	:	leetcode206.py
    @Author	:	fosaken
    @Time	:	2021/01/31 20:41:05
    @Abstract:

"""

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        cur_node = head
        last_head = None
        while cur_node is not None:
            cur_head = ListNode()
            cur_head.val = cur_node.val
            cur_head.next = last_head
            last_head = cur_head
            cur_node = cur_node.next
        return last_head